### PR TITLE
blog post draft

### PR DESCRIPTION
**Pull request by filippopedrazzinifp**

### CODE REVIEW

File: blog/2023-07-03-serve-llms-with-fastapi/index.md [hunk 1]
```diff
@@ -5,4 +5,384 @@ authors: [filippopedrazzinfp]
...
 
-![Prem Banner](./banner.png)
\ No newline at end of file
+![Prem Banner](./banner.png)
+
+<head>
+  <meta name="twitter:image" content="./banner.png"/>
+</head>
+
+In this tutorial, we are going to learn how to serve an LLM with FastAPI and Docker. For this tutorial I created a GitHub repostory with the code to serve `falcon-7b-instruct` already available [here](https://github.com/premAI-io/llm-fastapi-docker-template).
```
Comment (tiero) [inline blog/2023-07-03-serve-llms-with-fastapi/index.md:14]:
```suggestion
In this tutorial, we are going to learn how to prepare a LLM with FastAPI and Docker to be able to be served as Prem Service on the Prem platform. For this tutorial, I created a GitHub repostory with the code to serve `falcon-7b-instruct` already available [here](https://github.com/premAI-io/llm-fastapi-docker-template).
```

File: blog/2023-07-03-serve-llms-with-fastapi/index.md [hunk 2]
```diff
@@ -5,4 +5,384 @@ authors: [filippopedrazzinfp]
...
+
+> NOTE: in order to run Falcon 7B Instrcut model you will need a GPU with at least 16GiB of VRAM. You can use a [Paperspace Cloud](https://www.paperspace.com/gpu-cloud) virtual server or any other cloud provider or your own server with a NVIDIA GPU.
+
+### Step 1: Setup Microservice Dependencies
```
Comment (tiero) [inline blog/2023-07-03-serve-llms-with-fastapi/index.md:18]:
```suggestion
### Step 1: Setup the Python server
```

File: blog/2023-07-03-serve-llms-with-fastapi/index.md [hunk 3]
```diff
@@ -5,4 +5,384 @@ authors: [filippopedrazzinfp]
...
+httpx==0.23.3
+python-dotenv==1.0.0
+tenacity==8.2.2
+einops==0.6.1
+sentencepiece==0.1.99
+accelerate>=0.16.0,<1
+xformers==0.0.20
+```
+
+### Step 2: Expose the LLM with FastAPI
```
Comment (tiero) [inline blog/2023-07-03-serve-llms-with-fastapi/index.md:39]:
```suggestion
### Step 2: Expose the model with FastAPI
```

File: blog/2023-07-03-serve-llms-with-fastapi/index.md [hunk 4]
```diff
@@ -5,4 +5,384 @@ authors: [filippopedrazzinfp]
...
+
+COPY download.py .
+
+RUN python3 download.py --model $MODEL_ID
+
+COPY . .
+
+ENV MODEL_ID=$MODEL_ID
+
+CMD python3 main.py
```
Comment (tiero) [inline blog/2023-07-03-serve-llms-with-fastapi/index.md:347]: does buildx honor the cache without multi-stage Dockerfile?

**Review by tiero — COMMENTED** [review]
I would make the title and slug include prem and service:

`How to package an LLM to be served on Prem AI platform`

**Comment by filippopedrazzinifp** [comment]
@tiero I changed a few things. Please review it from scratch :)

File: blog/2023-07-03-serve-falcon-with-fastapi-and-docker/banner.png
Comment (biswaroop1547) [inline blog/2023-07-03-serve-falcon-with-fastapi-and-docker/banner.png:1]: @filippopedrazzinifp should we add the falcon x prem banner that was used in the twitter post here? or just the prem banner is fine?

Comment (filippopedrazzinifp) [reply blog/2023-07-03-serve-falcon-with-fastapi-and-docker/banner.png:1]: I will update it with another one. Soon ready.

**Pull request merged**